src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -115,40 +114,38 @@ void IslandWindow::Initialize()
     // stash the child interop handle so we can resize it when the main hwnd is resized
     interop->get_WindowHandle(&_interopWindowHandle);
 
-    if (!initialized)
-    {
-        _InitXamlContent();
-    }
-
+    _rootGrid = winrt::Windows::UI::Xaml::Controls::Grid();
     _source.Content(_rootGrid);
 
     // Do a quick resize to force the island to paint
     OnSize();
 }
 
-void IslandWindow::_InitXamlContent()
+double IslandWindow::GetCurrentDpiScale()
 {
     // setup a root grid that will be used to apply DPI scaling
     winrt::Windows::UI::Xaml::Media::ScaleTransform dpiScaleTransform;
     winrt::Windows::UI::Xaml::Controls::Grid dpiAdjustmentGrid;
 
     const auto dpi = GetDpiForWindow(_window);
     const double scale = double(dpi) / double(USER_DEFAULT_SCREEN_DPI);
-
-    _rootGrid = dpiAdjustmentGrid;
-    _scale = dpiScaleTransform;
-
-    _scale.ScaleX(scale);
-    _scale.ScaleY(scale);
+    return scale;
 }
 
 
 void IslandWindow::OnSize()
 {
     // update the interop window size
     SetWindowPos(_interopWindowHandle, 0, 0, 0, _currentWidth, _currentHeight, SWP_SHOWWINDOW);
-    _rootGrid.Width(_currentWidth);
-    _rootGrid.Height(_currentHeight);
+
+    if (_rootGrid)
+    {
+        const auto dpi = GetCurrentDpiScale();
+        const auto logicalWidth = (_currentWidth / dpi) + 0.5;

```

Comment:
Why the .5? Why not `ceil()` or something? What's going on here?

At a minimum the 0.5 needs to be explained in a comment.

Ideally I'd like all 4 of the usages to be the same function/algorithm instead of splatting the 0.5 in four places as well. Or perhaps a helper that puts in the height and width and assigns them to the root grid if it exists since this is close to the same code snippet up here and down below with one var changed.

Reply:
The 0.5 is to ensure that we pixel snap correctly at the edges, this is necessary with odd DPIs like 1.25,1.5,1,.75

I will document this and refactor the code